### PR TITLE
Add podcasts extension

### PR DESCRIPTION
**Pull request by ron-myers**
## Description

Add an initial `Podcasts` extension.

Commands are:
* `Play`
* `Pause`
* `Toggle Play/Pause`
* `Rewind 15 seconds`
* `Skip 30 seconds`

## Screencast

(I did not include one as all of the commands are `no-view` and interact with the Podcasts app.  If this approach is not helpful, just let me know I can can capture it running.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `media` folder

### CODE REVIEW

**Review by fe9lix — CHANGES_REQUESTED** [review]
Hey @ron-myers, thanks for your contribution. I've just tested the extension and it seems that the Apple Script for the shortcuts doesn't work for me. Here are the shortcut mappings for the Podcasts app on my machine:
CleanShot 2022-03-07 at 17 35 09@2x

For example, for the Play/Pause Toggle, it's just space, without option down...

File: extensions/podcasts/src/rewind.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { buildPodcastScript, runAppleScriptAndCloseWindow } from "./utils";
+
+export default async () => {
+  const playScript = `key code 123 using {shift down, command down}`; // shift, command, left arrow
```
Comment (fe9lix) [inline extensions/podcasts/src/rewind.ts:4]: Which key is code 123?

File: extensions/podcasts/src/utils.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      ${scriptToRunOnceActivated}
+      delay 0.1
+      set visible to true
+    end tell
+    activate application old
+`;
+}
+
+export async function togglePlayPause() {
+  const playScript = `key code 49 using {option down}`; // option + space
```
Comment (fe9lix) [inline extensions/podcasts/src/utils.ts:26]: On my machine, the toggle is just the space key...

File: extensions/podcasts/src/rewind.ts (hunk 1 above)
Comment (ron-myers) [reply extensions/podcasts/src/rewind.ts:4]: left arrow.

File: extensions/podcasts/src/utils.ts (hunk 2 above)
Comment (ron-myers) [reply extensions/podcasts/src/utils.ts:26]: Hrm - thank you for catching this @fe9lix.  Ok - key codes may not work if it is different by keyboard layout or culture.

Does your menu look like:
OnPaste 20220307-181409

**Comment by fe9lix** [comment]
Any other ways to make this more robust? Maybe triggering the menu items or control buttons directly? (Would probably require accessibility permissions, though).

File: extensions/podcasts/src/utils.ts (hunk 2 above)
Comment (ron-myers) [reply extensions/podcasts/src/utils.ts:26]: @fe9lix - can you confirm your menu looks the same as the image above (with different shortcut keys)?

Comment (fe9lix) [reply extensions/podcasts/src/utils.ts:26]: here's the menu on my machine:
CleanShot 2022-03-09 at 09 05 45@2x

Comment (ron-myers) [reply extensions/podcasts/src/utils.ts]: @fe9lix - I have updated to using the menu items.

Is there a way to note in an extension it requires accessibility?

**Comment by fe9lix** [comment]
@ron-myers In the latest version, I get the following error when using the "Play" command:

```
Error: Command failed with exit code 1: osascript -e 
    set old to (path to frontmost application as text)
    tell application "Podcasts"
      reopen
      activate
    end tell
    tell application "System Events" to tell process "Podcasts"
      
    tell menu bar item 6 of menu bar 1
      click
      click menu item 1 of menu 1
    end tell  

      delay 0.1
      set visible to false
    end tell
    activate application old

245:250: execution error: System Events got an error: Can’t get menu bar 1 of process "Podcasts". Invalid index. (-1719)

		error = new Error(message);
		        ^

Error: Command failed with exit code 1: osascript -e 
    set old to (path to frontmost application as text)
    tell application "Podcasts"
      reopen
      activate
    end tell
    tell application "System Events" to tell process "Podcasts"
      
    tell menu bar item 6 of menu bar 1
      click
      click menu item 1 of menu 1
    end tell  

      delay 0.1
      set visible to false
    end tell
    activate application old

245:250: execution error: System Events got an error: Can’t get menu bar 1 of process "Podcasts". Invalid index. (-1719)
makeError:error.js:60:11

---
57: 		error.originalMessage = error.message;
58: 		error.message = message;
59: 	} else {
60: 		error = new Error(message);
61: 	}
62: 
63: 	error.shortMessage = shortMessage;
---

handlePromise:index.js:118:26
processTicksAndRejections:task_queues:96:5
runAppleScript:index.js:9:19
runAppleScriptAndCloseWindow:utils.ts:6:3
togglePlayPause:utils.ts:37:3
Object.play_default:play.tsx:4:3
async MessagePort.<anonymous>:index.js:1:2278
```

**Comment by ron-myers** [comment]
> @ron-myers In the latest version, I get the following error when using the "Play" command:

Well, that is frustrating.  I will move to find the element by text.

@fe9lix - so I can better test before review, what are your Language and Region settings?  Can you think of any other reason our menu indexes would differ?  I am new to AppleScript.

**Comment by fe9lix** [comment]
Hey @ron-myers I've tested it again and interestingly, it works now!

**Review by fe9lix — APPROVED** [review]

**Pull request merged**